lib/internal/http2/core.js

```diff
@@ -3141,16 +3141,21 @@ function connect(authority, options, listener) {
   if (typeof listener === 'function')
     session.once('connect', listener);
 
-  debug('Http2Session connect', options.createConnection);
-  // Socket already has some buffered data - emulate receiving it
-  // https://github.com/nodejs/node/issues/35475
-  if (socket && socket.readableLength) {
-    let buf;
-    while ((buf = socket.read()) !== null) {
-      debug(`Http2Session connect: injecting ${buf.length} already in buffer`);
-      session[kHandle].receive(buf);
+  // Process the data on the next tick - a remoteSettings handler may be attached
+  // https://github.com/nodejs/node/issues/35981
+  process.nextTick(() => {
+    debug('Http2Session connect', options.createConnection);
+    // Socket already has some buffered data - emulate receiving it
+    // https://github.com/nodejs/node/issues/35475
+    if (socket && socket.readableLength) {
+      let buf;
+      while ((buf = socket.read()) !== null) {
+        debug(`Http2Session connect: injecting ${buf.length} already in buffer`);
```

Comment:
Splitting this onto two lines will make it pass the lint check or maybe this?

```suggestion
        debug(`Http2Session connect: ${buf.length} already in buffer`);
```

Maybe a add in a unit? (Would it be bytes?)

```suggestion
        debug(`Http2Session connect: ${buf.length} bytes already in buffer`);
```